include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,244 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+#define GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+
+#include <memory>
+#include <vector>
+
+#include <grpc/grpc_security.h>
+#include <grpc/support/log.h>
+#include <grpcpp/support/config.h>
+
+namespace grpc_impl {
+namespace experimental {
+
+/** TLS key materials config, wrapper for grpc_tls_key_materials_config. **/
+class TlsKeyMaterialsConfig {
+ public:
+  struct PemKeyCertPair {
+    grpc::string private_key;
+    grpc::string cert_chain;
+  };
+
+  /** Getters for member fields. **/
+  const grpc::string pem_root_certs() const { return pem_root_certs_; }
+  const std::vector<PemKeyCertPair>& pem_key_cert_pair_list() const {
+    return pem_key_cert_pair_list_;
+  }
+  int version() const { return version_; }
+
+  /** Setter for key materials that will be called by the user. The setter
+   * transfers ownership of the arguments to the config. **/
+  void set_key_materials(grpc::string pem_root_certs,
+                         std::vector<PemKeyCertPair> pem_key_cert_pair_list);
+  void set_version(int version) { version_ = version; };
+
+ private:
+  int version_ = 0;
+  std::vector<PemKeyCertPair> pem_key_cert_pair_list_;
+  grpc::string pem_root_certs_;
+};
+
+/** TLS credential reload arguments, wraps grpc_tls_credential_reload_arg. **/
+class TlsCredentialReloadArg {
+ public:
+  TlsCredentialReloadArg(grpc_tls_credential_reload_arg* arg) : c_arg_(arg) {}
```

Comment:
Please specify if the ownership of `arg` is transferred or not.

Reply:
Specified this for TlsCredentialReloadArg and for TlsServerAuthorizationCheckArg. Also added more comments on the setter functions and how they allocate memory.